amethyst_network/src/server/server_socket_event.rs

```diff
@@ -0,0 +1,44 @@
+use laminar::error::NetworkError;
+use laminar::{Event, Packet};
+
+/// Net event which occurred on the network.
+pub enum ServerSocketEvent {
+    /// event containing a packet with received data
+    Packet(Packet),
+    /// event containing an error that has occurred in the network
+    Error(NetworkError),
+    /// events that can happen with a client
+    ClientEvent(ClientEvent),
+    /// event used for defaults
+    Empty,
+}
+
+impl Clone for ServerSocketEvent {
+    fn clone(&self) -> Self {
+        ServerSocketEvent::Empty
+    }
+}
+
+/// Event that could occur with a client.
+pub enum ClientEvent {
+    /// represents a connecting client
+    Connected,
+    /// represents a disconnecting client
+    Disconnected,
+    /// represents a client who is timing out
+    Timedout,
+    /// represents an default value for this enum.
+    None,
```

Comment:
Can you explain in what situations you might get an event but its none? that seems redudant

Reply:
I was using that for events currently not implemented. But now I took a look at it I saw that it was only used for `QualityChange` event from laminar. I changed `None` to `QualityChange` now.